src/weapons.cpp

```diff
@@ -806,15 +815,16 @@ bool WeaponDistance::useWeapon(Player* player, Item* item, Creature* target) con
 	return true;
 }
 
-int32_t WeaponDistance::getElementDamage(const Player* player, const Creature* target, const Item* item) const
+int32_t WeaponDistance::getElementDamage(std::shared_ptr<const Player> player, std::shared_ptr<const Creature> target,
+                                         std::shared_ptr<const Item> item) const
 {
 	if (elementType == COMBAT_NONE) {
 		return 0;
 	}
 
 	int32_t attackValue = elementDamage;
 	if (item->getWeaponType() == WEAPON_AMMO) {
-		Item* weapon = player->getWeapon(true);
+		auto weapon = player->getWeapon(true);
```

Comment:
Could this be `const auto`
```suggestion
		cont auto weapon = player->getWeapon(true);
```